Setup.hs

```diff
@@ -1,2 +1,55 @@
-import Distribution.Simple
-main = defaultMain
+import           Data.Char
+import           Data.Maybe
+import           Distribution.PackageDescription
+import           Distribution.Simple
+import           Distribution.Simple.LocalBuildInfo
+import           Distribution.Simple.Program
+import           Distribution.Simple.Setup
+
+main :: IO ()
+main = defaultMainWithHooks simpleUserHooks
+  { hookedPrograms = map simpleProgram programs
+  , confHook       = configure
+  }
+
+programs :: [String]
+programs = ["hello"]
+
+configure
+  :: (GenericPackageDescription, HookedBuildInfo)
+  -> ConfigFlags
+  -> IO LocalBuildInfo
+configure (gpd, hbi) flags = do
+  lbi' <- confHook simpleUserHooks (gpd, hbi) flags
+  let paths = map
+        (\p ->
+          ( p
+          , maybe p
+                  programPath
+                  (lookupProgram (simpleProgram p) (withPrograms lbi))
+          )
+        )
+        programs
+      descr = localPkgDescr lbi'
+      lib   = fromJust (library descr)
+      libbi = libBuildInfo lib
+      lbi   = lbi'
+        { localPkgDescr = descr
+          { library = Just
+            (lib
+              { libBuildInfo = libbi
+                                 { cppOptions = cppOptions libbi
+                                                  ++ map fmtCppArg paths
```

Comment:
Finds the path of external programs to bake into the binary and pass that in through `cpp`